### PR TITLE
Fix event handling - Closes #4208

### LINKED ISSUE #4208: Handle transport endpoints from malicious payload
### Expected behavior
- If the payload is invalid, handle the exception and log accordingly and do not let it crash.
- Also, please fix the `Reason: [object Object]`
- Cover all the endpoint with a integration test, with positive and negative scenarios
### Actual behavior
When a user emits a malicious payload to transport endpoint, the application validates and throws an error, and the error not being handled causes the node to crash due to `unhandledRejection`
```
13:19:20.796Z DEBUG lisk-framework:
    Received post block broadcast request in unexpected format { errors:
       [ { keyword: 'type',
           dataPath: '',
           schemaPath: '#/type',
           params: [Object],
           message: 'should be object' } ],
      module: 'transport',
      query:
       '{"block":{"version":1,"reward":200000000,"payloadHash":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","timestamp":103574530,"previousBlock":"13940227048585692600","generatorPublicKey":"eaa049295d96618c51eb30deffe7fc2cc8bfc13190cb97f3b513dd060b000a46","blockSignature":"a41a2786df5b446de04f4d369481f3e126f20e87b50a6557fda3ddb5ccfc2390adfd1015284bceae653660af0dca2fa76bdfbcde97a48ea994923e89ec949400EXTRADATAHERETOCRASHTHENODE","height":9171510,"id":"8390758977495702841","relays":4,"totalAmount":0,"totalFee":0,"numberOfTransactions":0,"payloadLength":0,"transactions":[]},"nonce":"3WRejz1OirEgUb4Z"}' }
13:19:20.803Z FATAL lisk-framework:
    System error: unhandledRejection : { message: '[object Object]',
      stack:
       'Error: [object Object]\n    at Transport.postBlock (/home/lisk/lisk-main/node_modules/lisk-framework/src/modules/chain/transport/transport.js:366:10)\n    at Immediate.channel.subscribe (/home/lisk/lisk-main/node_modules/lisk-framework/src/modules/chain/chain.js:183:22)\n    at runCallback (timers.js:706:11)\n    at tryOnImmediate (timers.js:676:5)\n    at processImmediate (timers.js:658:5)\n    at process.topLevelDomainCallback (domain.js:126:23)' }
13:19:20.807Z  INFO lisk-framework: Cleaning network...
13:19:20.805Z  INFO lisk-framework: Cleaning chain...
13:19:20.805Z DEBUG lisk-framework: Cache - Clean up database
13:19:20.806Z DEBUG lisk-framework: Cache - Quit database
13:19:20.806Z  INFO lisk-framework: Cleaned up successfully
13:19:20.804Z  INFO lisk-framework: Cleanup controller...
13:19:20.804Z ERROR lisk-framework: Reason: [object Object]
```
### Steps to reproduce
- Create socket client and emit malformed payload to `postBlock` endpoint
### Which version(s) does this affect? (Environment, OS, etc...)
2.3.x

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
- network:event was subscribed, but it shouldn't throw an error

### How did I solve it?
- wrap with try-catch, and expect the error to happen. Also, it should log.

### How to manually test it?
- Use P2P library and call the endpoint with invalid data

### Review checklist

- [ ] The PR resolves #4208 
- [ ] All new code is covered with unit tests
- [ ] Relevant integration / functional tests are added
- [ ] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [ ] Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/chain.js [hunk 1]
```diff
@@ -170,21 +170,30 @@ module.exports = class Chain {
...
+							if (event === 'postSignatures') {
+								await this.transport.postSignatures(data);
+								return;
+							}
+							if (event === 'postBlock') {
+								await this.transport.postBlock(data);
+								return;
+							}
+						} catch (error) {
+							this.logger.info(
```
Comment (ishantiw) [inline framework/src/modules/chain/chain.js]: Shouldn't the log level of this error be `debug` or `error`?

Comment (ishantiw) [reply framework/src/modules/chain/chain.js]: maybe we can remove `data` to be logged as it clutters up the logs if someone constantly sends malformed data

Comment (shuse2) [reply framework/src/modules/chain/chain.js]: I thought error will be too much since it's handled properly, but user should know about those event.
Maybe info is too low on this case, so i increased it to warn

File: framework/src/modules/chain/chain.js [hunk 2]
```diff
@@ -170,21 +170,30 @@ module.exports = class Chain {
...
+								return;
+							}
+							if (event === 'postSignatures') {
+								await this.transport.postSignatures(data);
+								return;
+							}
+							if (event === 'postBlock') {
+								await this.transport.postBlock(data);
+								return;
+							}
```
Comment (ManuGowda) [inline framework/src/modules/chain/chain.js]: `return` at the end instead of returning each condition?

File: framework/src/modules/chain/transport/transport.js [hunk 3]
```diff
@@ -363,39 +363,24 @@ class Transport {
...
 			);
-			throw new Error(errors);
+			throw errors;
 		}
 
 		let block;
-		let success = true;
-		try {
-			block = blocksUtils.addBlockProperties(query.block);
+		block = blocksUtils.addBlockProperties(query.block);
```
Comment (ManuGowda) [inline framework/src/modules/chain/transport/transport.js]: why not just `let block = blocksUtils.addBlockProperties(query.block);`?

File: framework/src/modules/chain/chain.js (hunk 2 above)
Comment (shuse2) [reply framework/src/modules/chain/chain.js]: It should be fine, but I think in this case it's bit more clear it doesn't go to other cases

File: framework/src/modules/chain/transport/transport.js (hunk 3 above)
Comment (shuse2) [reply framework/src/modules/chain/transport/transport.js]: I would even do the `const block`, but since this is patch case and not to change a lot (because there is alot of those cases in this file), i would prefer not to touch it now

File: framework/src/modules/chain/chain.js (hunk 2 above)
Comment (diego-G) [reply framework/src/modules/chain/chain.js]: I agree with @ManuGowda. Using `else if` + 1 single return is cleaner.

File: framework/src/modules/chain/transport/transport.js [hunk 4]
```diff
@@ -363,39 +363,24 @@ class Transport {
...
-			success = false;
-			this.logger.debug('Block normalization failed', {
-				err: e.toString(),
-				module: 'transport',
-				block: query.block,
-			});
+		// Instantiate transaction classes
+		block.transactions = this.interfaceAdapters.transactions.fromBlock(block);
 
-			// TODO: If there is an error, invoke the applyPenalty action on the Network module once it is implemented.
```
Comment (diego-G) [inline framework/src/modules/chain/transport/transport.js]: Please, keep this TODO.

File: framework/test/mocha/functional/ws/transport/transport.js [hunk 5]
```diff
@@ -32,7 +32,7 @@ describe('WS transport', () => {
 		let p2p;
 
 		function postTransaction(transactionToPost) {
-			p2p.send({
+			return p2p.send({
```
Comment (diego-G) [inline framework/test/mocha/functional/ws/transport/transport.js]: Do we really need this return? I actually don't like the structure of the entire describe. I suggest to clean it:
- Its `beforeEach` is misleading
- Remove the use of done.
- The inner `describe` block is not necessary
- I would remove `postTransaction()` to ease the understanding of every test or move it to a more general handler. If you don't agree, we should follow this pattern later on and do the same thing when sending.

Comment (shuse2) [reply framework/test/mocha/functional/ws/transport/transport.js]: This PR is not intended to refactor the test.
All of the functional tests will be recovered in jest, so not needed to rewrite for this patch imo

File: framework/src/modules/chain/transport/transport.js (hunk 3 above)
Comment (shuse2) [reply framework/src/modules/chain/transport/transport.js]: ah nvm i think it's mistake will fix it
